sdk/storage/Azure.Storage.DataMovement/src/DownloadChunkHandler.cs

```diff
@@ -78,18 +90,29 @@ public struct Behaviors
         /// <param name="behaviors">
         /// Contains all the supported function calls.
         /// </param>
-        /// <param name="cancellationToken">
-        /// Optional <see cref="CancellationToken"/> to propagate
-        /// notifications that the operation should be cancelled.
-        /// </param>
         /// <exception cref="ArgumentException"></exception>
         public DownloadChunkHandler(
             long currentTransferred,
             long expectedLength,
             IList<HttpRange> ranges,
-            Behaviors behaviors,
-            CancellationToken cancellationToken)
+            Behaviors behaviors)
         {
+            // Create channel of finished Stage Chunk Args to update the bytesTransferred
+            // and for ending tasks like commit block.
+            // The size of the channel should never exceed 50k (limit on blocks in a block blob).
+            // and that's in the worst case that we never read from the channel and had a maximum chunk blob.
+            _downloadRangeChannel = Channel.CreateUnbounded<DownloadRangeEventArgs>(
+                new UnboundedChannelOptions()
+                {
+                    // Single reader is required as we can only read and write to bytesTransferred value
+                    SingleReader = true,
+                });
+            _channelCancellationSource = new CancellationTokenSource();
+            _processDownloadRangeEvents = Task.Run(() => NotifyOfPendingChunkDownloadEvents());
+
+            _expectedLength = expectedLength;
+            _ranges = ranges;
+
```

Comment:
Will cherry pick these changes out and put in a different PR for future review and testing.

https://github.com/Azure/azure-sdk-for-net/issues/32854